server/lib/actors/hot_test.go

```diff
@@ -0,0 +1,250 @@
+			tmp.addResultsWithFilter(e.results, func (ent *HotEntry) bool {
+				//log.Printf("addResultsWithFilter: %v %v %v %v %v %v", i, ent, begin, end, ent.within(begin, end), yue.DateByUUID(yuep.UUID(ent.id)))
+				return ent.within(begin, end)
+			})
+		}
+		//week/alltime: emurate filter on making 12h summery
+		if e.fetchAt.UnixNano() == lastFilter.UnixNano() && 
+			(typ == proto.TopicListRequest_Week || typ == proto.TopicListRequest_AllTime) {
+			log.Printf("emurate filter on summery %v %v %v", lastFilter, getClock(), tmp.total.entries)
+			tmp = filterSummery(btyp, tmp)
+			log.Printf("emurate filter on summery done %v", tmp.total.entries)
+		}
+	}
+
+	switch btyp {
+	case proto.TopicListRequest_Rising, proto.TopicListRequest_Hot:
+		//log.Printf("sort tmp: %v", tmp.total.entries)
+		tmp = tmp.sort(btyp).truncate(ENTRY_PER_FETCH)
+	case proto.TopicListRequest_Flame:
+		tmp = tmp.filter(func (e *HotEntry) bool {
+			return e.flamy()
+		}).sort(btyp).truncate(ENTRY_PER_FETCH)
+		//log.Printf("sort tmp: %v", tmp.total.entries)
+	}
+	last := int(start+count)
+	if last >= len(tmp.total.entries) {
+		last = len(tmp.total.entries)
+	}
+	//log.Printf("estimateResults %v %v %v %v", tmp.total.entries, start, count, last)
+	return tmp.total.entries[start:last]
+
+}
+
+func NewTestHotActor(locale string, updateSpan time.Duration, blen []int) (*HotActor, error) {
+	return NewHotActorWithDetail(locale, updateSpan, blen, true, testInmemFetcher, testInmemPersister);
+}
+
+func TestHotActor(t *testing.T) {
+log.Printf("current clock: %v", clock)
+	bucketTypes := []proto.TopicListRequest_BucketType {
+		proto.TopicListRequest_Rising,
+		proto.TopicListRequest_Hot,
+		proto.TopicListRequest_Flame,
+	}
+	queryTypes := []proto.TopicListRequest_QueryType {
+		proto.TopicListRequest_Hour,
+		proto.TopicListRequest_Day,
+		proto.TopicListRequest_Week,
+		proto.TopicListRequest_AllTime,
+	}
+	blen := []int{2, 24}
+	// set 4th argument to true for calling update manually.
+	a, _ := NewTestHotActor("jp", 1 * time.Second, blen);
+	for i := 0; i < ITER_COUNT; i++ {
+		addClock()
+		a.update(getClock())
+		a.updateQueryCache(getClock())
+		start, count := 0, 5
+		btyp := bucketTypes[rand.Int31n(int32(len(bucketTypes)))]
+		var qtypIndex int32
+		if btyp == proto.TopicListRequest_Rising {
+			//Rising only has Hour and Day
+			qtypIndex = rand.Int31n(int32(len(queryTypes) - 2))
+		} else {
+			qtypIndex = rand.Int31n(int32(len(queryTypes)))			
+		}
+		typ := queryTypes[qtypIndex]
+		log.Printf("iter: %v %v %v", i, btyp, typ)
+		entries, err := a.Query(btyp, typ, nil, 0, 0, uint32(count))
+		if err != nil {
+			t.Fatal(err.Error())
+		} else {
+			estEntries := estimateResults(btyp, typ, uint64(start), uint64(count))
+			log.Printf("check: \n%v\n%v", entries, estEntries)
+			if len(entries) != len(estEntries) {
+				t.Fatal("entry length differ %v %v", entries, estEntries)
```

Comment:
__[go_vet]__
* [vet] possible formatting directive in Fatal call

<!-- Comment by [SideCI](https://sideci.com) -->